<launch>
    <arg name="camera_name" default="decklink"/>
    <arg name="calibration_url" default=""/>
    <arg name="frame_id" default=""/>

    <arg name="x_offset" default="420"/>
    <arg name="y_offset" default="0"/>
    <arg name="width" default="1080"/>
    <arg name="height" default="1080"/>

    <group ns="decklink">
        <!-- launch decklink_ros -->
        <node pkg="decklink_ros" name="decklink_ros_node" type="decklink_ros_node" output="screen">
            <param name="camera_name" type="str" value="$(arg camera_name)"/>
            <param name="calibration_url" type="str" value="$(arg calibration_url)"/>
            <param name="frame_id" type="str" value="$(arg frame_id)"/>
        </node>

        <!-- see https://answers.ros.org/question/248194/how-to-use-image_proccrop_decimate-to-crop-an-image/ -->
        <node pkg="nodelet" name="image_proc_crop_decimate" type="nodelet" args="standalone image_proc/crop_decimate">
            <param name="x_offset" type="int" value="$(arg x_offset)"/>
            <param name="y_offset" type="int" value="$(arg y_offset)"/>
            <param name="width" type="int" value="$(arg width)"/>
            <param name="height" type="int" value="$(arg height)"/>

            <remap from="camera/image_raw" to="image_raw"/>
            <remap from="camera/camera_info" to="camera_info"/>

            <remap from="camera_out/image_raw" to="crop/image_raw"/>
            <remap from="camera_out/camera_info" to="crop/camera_info"/>
        </node>

        <!-- note that the image is rectified as part of the image processing pipelin in h_vs -->
    </group>

</launch>
    